framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1506,6 +1507,55 @@ private Object getValueOfExpressionInAnnotation(
     }
 
     /**
+     * Returns the TypeElement with the fully qualified name {@code name}, if one exists. Otherwise,
+     * checks {@code enclosingClass} and {@code packageName} for such a class with {@code name}.
+     *
+     * @param packageName name of package being parsed
+     * @param enclosingClass fully-qualified name of the class being parsed
+     * @param name classname (Simple or fully qualified)
+     * @return the TypeElement for {@code name}
+     */
+    private TypeElement findTypeOfName(String packageName, String enclosingClass, String name) {
+        TypeElement typeElement = elements.getTypeElement(name);
```

Comment:
It's a bit annoying that these 5 lines are repeated 4 times. How about rewriting it as:

```java
private TypeElement checkCandidate(String name) {
    TypeElement typeElement = elements.getTypeElement(name);
     if (typeElement != null) {
         importedTypes.put(name, typeElement);
     }
     return typeElement;
}

private TypeElement findTypeOfName(String packageName, String enclosingClass, String name) {
    TypeElement typeElement = checkCandidate(name);
    while (typeElement == null && !enclosingClass.equals(packageName)) {
        typeElement = checkCandidate(enclosingClass + "." + name);
        int lastDot = enclosingClass.lastIndexOf('.');
        if (lastDot == -1) {
            break;
        } else {
            enclosingClass = enclosingClass.substring(0, lastDot);
        }
    }
    if (typeElement == null && !packageName.isEmpty()) {
        typeElement = checkCandidate(packageName + "." + name);
    }
    if (typeElement == null && !packageName.equals("java.lang")) {
        typeElement = checkCandidate("java.lang." + name);
    }
    return typeElement;
}
````

Maybe a few more null checks, but less repetition and only a single return.